src/js/index.js

```diff
@@ -225,6 +233,25 @@ export const hpe = deepFreeze({
     extend: css`
       ${props => !props.plain && 'font-weight: bold;'}
     `,
+    disabled: {
+      border: {
+        color: 'disabled-border',
+      },
+      color: 'disabled-text',
+      opacity: 1.0,
+      extend: ({ primary, theme }) =>
```

Comment:
Any reason this can't be inside the `primary` part of the theme, rather than having to test for that here?

Reply:
Since primary styling typically uses color as the fill for the button, I had set up primary disabled to use the disabled color as the fill for primary buttons and label color otherwise. 

Since designs call for uniform fill/label color regardless of button type, I did it this way. However, if we don't want `color` to control background fill for primary buttons we can discuss and update how grommet does the styling.

Reply:
In general, I'm trying have the theme not need to inspect props. To me, that feels like the theme structure is missing something. Here's the general theme structure approach:
```
button.primary.background
button.disabled.primary.background
```
So, we start with non-state keys and then we add top level keys for each state and replicate the non-state keys as needed under it.

Reply:
Great, I think this is a more consistent convention. I will look into updating this on the Grommet side of things.